tests/Interface/core/ImageContainerTest.cpp

```diff
@@ -24,6 +24,38 @@
 #include <buddy/Core/Container.h>
 #include <buddy/DIP/ImageContainer.h>
 
+bool compare_flt(float a, float b) {
+  if (a == b)
+    return true;
+  float eps = std::abs(a - b);
+  return (eps < FLT_EPSILON);
+}
+
+template <typename T, size_t N>
+bool testImgcvnorm(cv::Mat testImgcv, Img<T, N> testImg, bool norm = false) {
+  int cvn = testImgcv.dims;
+  if (cvn != N)
+    return false;
+  for (size_t i = 0; i < N; i++) {
+    if (testImgcv.size[i] != testImg.getSizes()[i])
+      return false;
+  }
+  T *data = testImg.getData();
+  if (N == 2) {
```

Comment:
Can you make this more generic instead of only handling for 2 dimensions?

Reply:
can you more give more information , generic as in for N>3 
but as i see at Imagecontainer.h ,for (N!=2)  cv image is not handled when N==3 / for NCHW layout i see copying only first channel information of image than number of channels 

can you give me direction /help on what can be done

Reply:
Can you support testing with the NCHW layout? (and add a TODO here to note that tests should be updated there once more layouts are supported functionally)

Reply:
can you please check now